EtlPipelineBlobTrigger/course_docs.py

```diff
@@ -124,22 +110,58 @@ def get_links(locations, locids, raw_inst_data, raw_course_data):
     return links
 
 
-def get_location_items(locations, locids, raw_inst_data, raw_course_data):
+def get_location_items(locations, locids, raw_course_data, pub_ukprn):
+
+    course_locations = SharedUtils.get_raw_list(raw_course_data,
+                                                "COURSELOCATION")
+    item = {}
+    for course_location in course_locations:
+        if 'UCASCOURSEID' in course_location:
+            item[course_location['LOCID']+pub_ukprn] = course_location[
+                                                        'UCASCOURSEID']
+
     location_items = []
     for locid in locids:
         location_dict = {}
         raw_location_data = locations.get_location_data_for_key(locid)
+
+        if raw_location_data is None:
+            # TODO Handle gracefully if unable to find lookup
+            logging.warning(f'failed to find location data in lookup table')
```

Comment:
Perhaps continue if only want to log a warning. If you carry on, then it will raise an error as soon as you try to iterate into raw_location_data below.